pymc3/distributions/mixture.py

```diff
@@ -612,3 +618,183 @@ def __init__(self, w, mu, sigma=None, tau=None, sd=None, comp_shape=(), *args, *
+        with shape (batch_shape, mixture_axis, event_shape)
+    mixture_axis: int
+        Axis to be reduced in the mixture
+    """
+
+    def __init__(self, w, comp_dists, mixture_axis=-1, *args, **kwargs):
+        self.w = tt.as_tensor_variable(w)
+        if not isinstance(comp_dists, Distribution):
+            raise TypeError(
+                "The MixtureSameFamily distribution only accepts Distribution "
+                f"instances as its components. Got {type(comp_dists)} instead."
+            )
+        self.comp_dists = comp_dists
+        if mixture_axis < 0:
+            mixture_axis = len(comp_dists.shape) + mixture_axis
+        comp_shape = to_tuple(comp_dists.shape)
+        self.shape = comp_shape[:mixture_axis] + comp_shape[mixture_axis + 1 :]
+        self.mixture_axis = mixture_axis
+        kwargs.setdefault("dtype", self.comp_dists.dtype)
+
+        # Computvalte the mode so we don't always have to pass a tes
+        defaults = kwargs.pop("defaults", [])
+        event_shape = self.comp_dists.shape[mixture_axis + 1 :]
+        _w = tt.shape_padleft(
+            tt.shape_padright(w, len(event_shape)),
+            len(self.comp_dists.shape) - w.ndim - len(event_shape),
+        )
+        mode = take_along_axis(
+            self.comp_dists.mode,
+            tt.argmax(_w, keepdims=True),
+            axis=mixture_axis,
+        )
+        self.mode = mode[(..., 0) + (slice(None),) * len(event_shape)]
+        defaults.append("mode")
+
+        super().__init__(defaults=defaults, *args, **kwargs)
+
+    def logp(self, value):
+        """
+        Calculate log-probability of defined ``MixtureSameFamily`` distribution at specified value.
+
+        Parameters
+        ----------
+        value : numeric
+            Value(s) for which log-probability is calculated. If the log probabilities for multiple
+            values are desired the values must be provided in a numpy array or theano tensor
+
+        Returns
+        -------
+        TensorVariable
+        """
+        # self.w.shape (batch_shape, mixture_axis)
+        # self.comp_dists.shape (batch_shape, mixture_axis, event_shape)
+        # value.shape (batch_shape, event_shape)
+
+        comp_dists = self.comp_dists
+        w = self.w
+        mixture_axis = self.mixture_axis
+
+        event_shape = comp_dists.shape[mixture_axis + 1 :]
+
+        # To be able to broadcast the comp_dists.logp with w and value
+        # We first have to pad the shape of w to the right with ones
+        # so that it can broadcast with the event_shape.
+
+        w = tt.shape_padright(w, len(event_shape))
+
+        # Second, we have to add the mixture_axis to the value tensor
+        # To insert the mixture axis at the correct location, we use the
+        # negative number index. This way, we can also handle situations
+        # in which, value is an observed value with more batch dimensions
+        # than the ones present in the comp_dists
```

Comment:
```suggestion
        # Second, we have to add the `mixture_axis` to the `value` tensor.
        # To insert the mixture axis at the correct location, we use the
        # negative number index. This way, we can also handle situations
        # in which `value` is an observed value with more batch dimensions
        # than the ones present in the `comp_dists`.
```

Reply:
I don't think that we should add markdown monospace delimiters in the comments. They wont be rendered anywhere. They're just there to help other developers to understand what's going on.